# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/media/platforms/wrappers/MediaChangeMonitor.cpp
# Commit: fd44d547e093
# Full Hash: fd44d547e0931ddcb25e9dde2691ad9e99ec2e05
# Author: alwu <alwu@mozilla.com>
# Date: 2023-10-26 09:13:45
# Description:
#   Bug 1860840 - return error for invalid hvcc config when converting sample. r=media-playback-reviewers,padenot
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D191794
# ==============================================================================

diff -r 691148c77012 -r fd44d547e093 dom/media/platforms/wrappers/MediaChangeMonitor.cpp
--- a/dom/media/platforms/wrappers/MediaChangeMonitor.cpp	Wed Oct 25 21:14:33 2023 +0300
+++ b/dom/media/platforms/wrappers/MediaChangeMonitor.cpp	Wed Oct 25 17:10:46 2023 +0000
@@ -189,12 +189,12 @@
 
   MediaResult CheckForChange(MediaRawData* aSample) override {
     // To be usable we need to convert the sample to 4 bytes NAL size HVCC.
-    if (!AnnexB::ConvertSampleToHVCC(aSample)) {
+    if (auto rv = AnnexB::ConvertSampleToHVCC(aSample); rv.isErr()) {
       // We need HVCC content to be able to later parse the SPS.
       // This is a no-op if the data is already HVCC.
       nsPrintfCString msg("Failed to convert to HVCC");
       LOG("%s", msg.get());
-      return MediaResult(NS_ERROR_OUT_OF_MEMORY, msg);
+      return MediaResult(rv.unwrapErr(), msg);
     }
 
     if (!AnnexB::IsHVCC(aSample)) {